src/Compilers/CSharp/Portable/Binder/WithLambdaParametersBinder.cs

```diff
@@ -92,7 +95,10 @@ internal override TypeWithAnnotations GetIteratorElementType(YieldStatementSynta
 
             foreach (var parameterSymbol in parameterMap[name])
             {
-                result.MergeEqual(originalBinder.CheckViability(parameterSymbol, arity, options, null, diagnose, ref useSiteDiagnostics));
+                if (!parameterSymbol.IsDiscard)
```

Comment:
>if (!parameterSymbol.IsDiscard) [](start = 16, length = 31)

It looks like this check is redundant, we never add discard parameters to this map.